### ISSUE TITLE
Steam Deck client launched from desktop asks for repeated login until opened with `-gamepadui`, or until OOBE is completed in gamescope-session

### ISSUE THREAD

**Issue by matte-schwartz** [ValveSoftware/steam-for-linux#11419]
#### Your system information

* Steam client version (build number or date): 1714854927
* Distribution (e.g. Ubuntu): Arch Linux
* Opted into Steam client beta?: N/A
* Have you checked for system updates?: N/A
* Steam Logs: [logs.tar.gz](https://github.com/user-attachments/files/17612395/logs.tar.gz)
* GPU: AMD
* System Report: [system-report.txt](https://github.com/user-attachments/files/17612459/system-report.txt)

#### Please describe your issue in as much detail as possible:
When launching the Steam Deck client (`steam -steamdeck`) on desktop, the Steam client will prompt you to log-in with a QR code repeatedly if you only launch it with that single parameter. It will start to remember your log-in details once launched with the addition of `-gamepadui`. 

#### Steps for reproducing this issue:

Starting from a vanilla Arch setup (used EndeavourOS for convenience), I grabbed the Steam jupiter bootstrap from https://steamdeck-packages.steamos.cloud/archlinux-mirror/jupiter-main/os/x86_64/steam-jupiter-stable-1.0.0.79-1-x86_64.pkg.tar.zst and installed it. After this, I followed the steps below before grabbing my Steam logs while on Plasma 6 Wayland:

1. `steam -steamdeck` -> had to log-in -> quit
2. `steam -steamdeck` -> had to log-in again -> quit
3. `steam -steamdeck -gamepadui` -> it knew my steam account already and let me click my profile ->loaded into library -> quit
4. `steam -steamdeck -gamepadui` -> straight to my library page in gamepadui -> quit
5. `steam -steamdeck` -> straight to my desktop library page

So it seems like something between steps 2 and 3 is what actually made the client remember my account and stop prompting for a log-in. The logs folder above was grabbed after Step 5, so it should be easy to keep track of each entry in my logs.

Another way to work around this issue is by installing all packages necessary to run gamescope-session and then completing the OOBE workflow in gamescope-session.

**Comment by kisak-valve** [comment]
Hello @matte-schwartz, we'll need to hear from a Steam dev, but there's a decent chance that this particular quirk is a won't fix because you've gone out of your way to run the tailored-for-Steam-Deck Steam client variant outside of the environment it's paired with. Using the `-steamdeck` runtime option comes with the expectation that the Steam client is being run in an environment similar to SteamOS's game mode.

**Comment by matte-schwartz** [comment]
Yep, thanks @kisak-valve. fully aware it's many degrees of unsupported and didn't report it for exactly that reason for a while. This was following up on a comment from a Steam dev about this topic from outside of GitHub, and wanted to share my findings and logs from it somewhere trackable. No problem if it ends up being a close/won't fix

**Comment by uzolotikov** [comment]
Exactly the same issue on steam deck itself (remember credentials in game mode and forget them in desktop). Any updates on the issue?

**Comment by matte-schwartz** [comment]
this no longer seems to be an issue on recent updates of the Steam client. I was able to install and use `steam-jupiter-stable-1.0.0.85` on Arch with `steam -steamdeck` on desktop mode and it remembered my login on the first try without invoking `-gamepadui`.
